core/remotes/resolver.go

```diff
@@ -109,3 +113,77 @@ func WithMediaType(mediatype string) FetchByDigestOpts {
 		return nil
 	}
 }
+
+////
+// Layer fetch config
+////
+
+type FetcherConfig struct {
```

Comment:
does this need to be exported?

Reply:
Yes, it's passed in `core/remotes/docker.` by `core/remotes` to pass the download settings/limitations.